src/macros.jl

```diff
@@ -283,6 +283,7 @@ end
 #constructconstraint!(x::Array, sense::Symbol) = map(c->constructconstraint!(c,sense), x)
 #constructconstraint!(x::AbstractArray, sense::Symbol) = constructconstraint!([x[i] for i in eachindex(x)], sense)
 
+constructconstraint!(x::AbstractArray, set::MOI.AbstractScalarSet) = error("Dot operators should be used to compare array expressions.")
```

Comment:
- This could be more helpful, e.g., ``"Unexpected vector in scalar constraint. Did you mean to use the dot comparison operators like .==, .<=, and .>= instead?"``

- Does this also trigger for `[x,y] in LessThan(1.0)`? If so, the error message is not helpful at all.

Reply:
With your suggestion of error message, it is clear that it is just a tip for common mistake so the case `[x,y] in LessThan(1.0)` is not an issue.